workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/ComponentDiffStorageGenerator.tt

```diff
@@ -0,0 +1,187 @@
+    var generatedHeader = CommonGeneratorUtils.GetGeneratedHeader();
+    var commandDetailsList = GetCommandDetailsList();
+    var eventDetailsList = GetEventDetailsList();
+    var componentNamespace = qualifiedNamespace + "." + componentDetails.ComponentName;
+    var profilingStart = $"Profiler.BeginSample(\"{componentDetails.ComponentName}\");";
+    var profilingEnd = "Profiler.EndSample();";
+#>
+<#= generatedHeader #>
+
+using System.Collections.Generic;
+using System.Linq;
+using Unity.Mathematics;
+using Unity.Entities;
+using Improbable.Gdk.Core;
+using Improbable.Worker.CInterop;
+
+namespace <#= qualifiedNamespace #>
+{
+    public partial class <#= componentDetails.ComponentName #>
+    {
+        public class DiffComponentStorage : IDiffUpdateStorage<Update>, IDiffComponentAddedStorage<Update>, IDiffAuthorityStorage
+<# foreach (var ev in eventDetailsList) {
+        var eventType = ev.EventName + ".Event";
+#>
+            , IDiffEventStorage<<#= eventType #>>
+<# } #>
+        {
+            private readonly HashSet<EntityId> entitiesUpdated = new HashSet<EntityId>();
+
+            private List<EntityId> componentsAdded = new List<EntityId>();
+            private List<EntityId> componentsRemoved = new List<EntityId>();
+
+            // Used to represent a state machine of authority changes. Valid state changes are:
+            // authority lost -> authority lost temporarily
+            // authority lost temporarily -> authority lost
+            // authority gained -> authority gained
+            // Creating the authority lost temporarily set is the aim as it signifies authority epoch changes
+            private readonly HashSet<EntityId> authorityLost = new HashSet<EntityId>();
+            private readonly HashSet<EntityId> authorityGained = new HashSet<EntityId>();
+            private readonly HashSet<EntityId> authorityLostTemporary = new HashSet<EntityId>();
+
+            private ReceivedMessageList<ComponentUpdateReceived<Update>> updateStorage =
+                new ReceivedMessageList<ComponentUpdateReceived<Update>>();
+
+            private ReceivedMessageList<AuthorityChangeReceived> authorityChanges =
+                new ReceivedMessageList<AuthorityChangeReceived>();
+<# foreach (var ev in eventDetailsList) {
+        var eventType = ev.EventName + ".Event";
+#>
+
+            private ReceivedMessageList<ComponentEventReceived<<#= eventType #>>> <#= ev.CamelCaseEventName #>EventStorage =
+                new ReceivedMessageList<ComponentEventReceived<<#= eventType #>>>();
+<# } #>
+
+            public uint GetComponentId()
+            {
+                return ComponentId;
+            }
+
+            public void Clear()
+            {
+                entitiesUpdated.Clear();
+                updateStorage.Clear();
+                authorityChanges.Clear();
+                componentsAdded.Clear();
+                componentsRemoved.Clear();
+<# foreach (var ev in eventDetailsList) {#>
+
```

Comment:
nite: whitespace not needed

Reply:
This makes the generated code easier to read.

Reply:
Misread which line this was on. Fixed now.